--- watchdog-5.13.orig/redhat/watchdog.init     2014-11-12 17:18:39.125943000 +0100
+++ watchdog-5.13/redhat/watchdog-init	2017-03-06 12:49:31.701240043 +0100
@@ -30,11 +30,12 @@
 		echo
 		return 1
 	fi
-	if [ "$VERBOSE" = "yes" ]; then
-	    /usr/sbin/${prog} -v
-	else
-	    /usr/sbin/${prog}
-        fi
+	#if [ "$VERBOSE" = "yes" ]; then
+	#    /usr/sbin/${prog} -v
+	#else
+	#    /usr/sbin/${prog}
+        #fi
+	/etc/init.d/startwatchdog $VERBOSE &
 	RETVAL=$?
 	[ $RETVAL -eq 0 ] && touch $lockfile
 	[ $RETVAL -eq 0 ] && success

